test/SignalRServiceExtension.Tests/Config/ServiceManagerStoreTests.cs

```diff
@@ -0,0 +1,81 @@
+    {
+        [Fact]
+        [Obsolete]
+        public void GetServiceManager_WithSingleEndpoint()
+        {
+            var connectionString = FakeEndpointUtils.GetFakeConnectionString(1).Single();
+            var configuration = new ConfigurationBuilder().AddInMemoryCollection().Build();
+            var connectionStringKey = "key";
+            configuration[connectionStringKey] = connectionString;
+
+            var managerStore = new ServiceManagerStore(configuration, NullLoggerFactory.Instance, null);
+            var hubContextStore = managerStore.GetOrAddByConnectionStringKey(connectionStringKey);
+            var manager = hubContextStore.ServiceManager;
+        }
+
+        [Fact]
+        [Obsolete]
+        public void GetServiceManager_WithMultipleEndpoints()
+        {
+            var connectionStrings = FakeEndpointUtils.GetFakeConnectionString(2).ToArray();
+            var configuration = new ConfigurationBuilder().AddInMemoryCollection().Build();
+            var connectionStringKey = "key";
+            configuration[connectionStringKey] = connectionStrings[0];
+            configuration[$"{connectionStringKey}:abc"] = connectionStrings[1];
+
+            var managerStore = new ServiceManagerStore(configuration, NullLoggerFactory.Instance, null);
+            var hubContextStore = managerStore.GetOrAddByConnectionStringKey(connectionStringKey);
+            Assert.Throws<InvalidOperationException>(() => hubContextStore.ServiceManager);
+        }
+
+        [Fact]
+        public void ProductInfoExists()
+        {
+            var connectionString = FakeEndpointUtils.GetFakeConnectionString(1).Single();
+            var configuration = new ConfigurationBuilder().AddInMemoryCollection().Build();
+            var connectionStringKey = "key";
+            configuration[connectionStringKey] = connectionString;
+
+            var productInfo = new ServiceCollection()
+                .AddSignalRServiceContext(configuration, connectionStringKey, null)
+                .BuildServiceProvider()
+                .GetRequiredService<IOptions<ContextOptions>>()
+                .Value.ProductInfo;
+            Assert.NotNull(productInfo);
+        }
+
+        [Fact]
+        public async Task DefaultRouterExists()
+        {
+            var builder = new HostBuilder();
+            var host = builder
+                .ConfigureAppConfiguration(b => b.AddInMemoryCollection(new Dictionary<string, string> { {"key", FakeEndpointUtils.GetFakeConnectionString(1).Single()
+        } }))
+                .ConfigureWebJobs(b => b.AddSignalR()).Build();
+            var hubContextStore = host.Services.GetRequiredService<IServiceManagerStore>().GetOrAddByConnectionStringKey("key");
+            var exp = await Assert.ThrowsAsync<AzureSignalRException>(()=> hubContextStore.ServiceContext.GetClientEndpointAsync("hubName"));
+            Assert.IsType<AzureSignalRNotConnectedException>(exp.InnerException);
```

Comment:
why `AzureSignalRException` has a inner exception that is also a `AzureSignalRException`